docker/docker_client.go

```diff
@@ -229,7 +230,7 @@ func newDockerClientWithDetails(sys *types.SystemContext, registry, username, pa
 // CheckAuth validates the credentials by attempting to log into the registry
 // returns an error if an error occcured while making the http request or the status code received was 401
 func CheckAuth(ctx context.Context, sys *types.SystemContext, username, password, registry string) error {
-	newLoginClient, err := newDockerClientWithDetails(sys, registry, username, password, "", nil, "")
+	newLoginClient, err := newDockerClientWithDetails(sys, registry, "", nil, "")
```

Comment:
How does this work? `username` and `password` are now ignored, and validating the two is IIRC the whole purpose of `CheckAuth`.

Reply:
They are not ignored. `newDockerClientWithDetails` calls `GetAuthentication` and retrieves the username and password. Initially `newDockerClientFromRef` was doing the get, but `newDockerClientFromRef` is not used anywhere in the search path. Plus `newDockerClientFromRef` calls `newDockerClientWithDetails` so doing the `GetAuthentication` in there makes more sense.

Reply:
> newDockerClientWithDetails calls GetAuthentication and retrieves the username and password.

Retrieves them _from the auth file_.

Compare with https://github.com/projectatomic/libpod/blob/c08a1e0b1124cf0b7c4bb80e09717134540ec593/cmd/podman/login.go#L83 , which first calls `CheckAuth`, and only on success writes the credentials to the auth file.

Reply:
(by “`username` and `password` are now ignored” I refer to the parameters to `CheckAuth`, not to `newDockerClient*`)

Reply:
ohh, I see it now. My bad, fixing...